### ISSUE TITLE
Be more verbose on filename errors

### ISSUE THREAD

**Issue by m3nu** [vsespb/mt-aws-glacier#34]
I just started uploading a big directory with chaotic filenames and got some invalid filenames. It will throw an "Invalid filename"-error, but no hint on which file is affected. In the end I added some print-statements to hunt it down.

**Comment by vsespb** [comment]
Hello.

There is error, which thrown in case filename is invalid (when it contains non-UTF characters):

```
                print STDERR "=== ===\nERROR: file/dir with invalid characters\nDump of file name bytes:\n"
                require Devel::Peek;
                Devel::Peek::Dump($binaryfilename);
                print STDERR "=== ===\n";
                croak;
```

it prints problem characters perfectly visible.

And there are errors in lines like this

```
 confess "invalid filename" unless
```

it should never happen actually. or happen in case of programming error, or in case something very non standard.

Than is why `confess` used - it prints stacktrace, and it does not really have to be user friendly.

However if you give me example what actually happened - what the problem was with filename, and on what line(there are several) I will rewrite it with normal, user friendly error message.

**Comment by m3nu** [comment]
I had 2 files failing in about 10k files. Unfortunately I already renamed them. I will check the backup for the files to reproduce it.

I encountered the confess-error (second example you gave).

**Comment by vsespb** [comment]
> I will check the backup for the files to reproduce it.

Yes, that would be great. Thanks.

> I encountered the confess-error (second example you gave).

There are 5 similar. Well, I believe you probably had last one.

```
grep 'invalid filename' Journal.pm

                confess "invalid filename" unless defined($relfilename = sanity_relative_filename($relfilename));
                confess "invalid filename" unless defined($relfilename = sanity_relative_filename($relfilename));
                confess "invalid filename" unless defined(my $filename = sanity_relative_filename($e->{relfilename}));
                confess "invalid filename" unless defined(my $filename = sanity_relative_filename($e->{relfilename}));
                                                confess "invalid filename" unless defined($relfilename = sanity_relative_filename($relfilename));
```

**Comment by m3nu** [comment]
The one in line 258.

**Comment by vsespb** [comment]
So, I don't know what exactly was problem with filename, but, fixed in 0.952beta:
1. There are now sane error messages for filenames containing CR, LF, or TAB characters.
2. For other problems with filename, there is a stacktrace (unexpected error) and filename dump (so user will be able to find this file)
3. Documentation added ("Limitation" section) with filenames limitations.
